sanic/app.py

```diff
@@ -637,14 +637,22 @@ def url_for(self, view_name: str, **kwargs):
 
         if external:
             if not scheme:
-                if ":" in netloc[:8]:
+                if netloc and ":" in netloc[:8]:
                     scheme = netloc[:8].split(":", 1)[0]
                 else:
                     scheme = "http"
 
-            if "://" in netloc[:8]:
+            if netloc and "://" in netloc[:8]:
                 netloc = netloc.split("://", 1)[-1]
 
+            if not netloc:
+                scheme = ""
+                external = False
+
+        port = kwargs.pop("_port", None)
+        if port is None:
+            port = self.config.get("SERVER_PORT", "")
```

Comment:
Wouldn't it be much simpler to merge all of these into a single line and let python handle it rather than the conditional branching?

```python
port = kwargs.pop('_port', self.config.get('SERVER_PORT', ''))
```

Reply:
single line format will not get the value from configuration if someone passes `_port=None` as parameter.

Reply:
@lixxu 
The following should give you the same output.

```python
port = kwargs.pop('_port', self.config.get('SERVER_PORT', '')) or ''
```

Reply:
Just FYI, 

```bash
>>> timeit.timeit(stmt=lookup, number=1000000)
0.472626347996993
>>> timeit.timeit(stmt=cond, number=1000000)
0.5024505830078851
```

Reply:
ok, updated for it. but I made the commits history ugly. :( wonder how to squash it again.